sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusReceiverClientIntegrationTest.java

```diff
@@ -0,0 +1,477 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.servicebus;
+
+import com.azure.core.util.IterableStream;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.messaging.servicebus.implementation.DispositionStatus;
+import com.azure.messaging.servicebus.implementation.MessagingEntityType;
+import com.azure.messaging.servicebus.models.ReceiveAsyncOptions;
+import com.azure.messaging.servicebus.models.ReceiveMode;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import reactor.test.StepVerifier;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Integration tests for {@link ServiceBusReceiverClient} from queues or subscriptions.
+ */
+public class ServiceBusReceiverClientIntegrationTest extends IntegrationTestBase {
+
+    private final ClientLogger logger = new ClientLogger(ServiceBusReceiverClientIntegrationTest.class);
+
+    ServiceBusReceiverClient receiver;
+    ServiceBusSenderClient sender;
+    /**
+     * Receiver used to clean up resources in {@link #afterTest()}.
+     */
+    ServiceBusReceiverClient receiveAndDeleteReceiver;
+
+    final AtomicInteger messagesPending = new AtomicInteger();
+
+
+    protected ServiceBusReceiverClientIntegrationTest() {
+        super(new ClientLogger(ServiceBusReceiverClientIntegrationTest.class));
+    }
+
+    @BeforeAll
+    static void beforeAll() {
+        StepVerifier.setDefaultTimeout(Duration.ofSeconds(30));
```

Comment:
You could probably move this BeforeAll and AfterAll setup into the base class because we want there to be a timeout.